AEPIdentity/Sources/IdentityState.swift

```diff
@@ -36,33 +37,51 @@ class IdentityState {
         self.pushIdManager = pushIdManager
     }
 
-    /// Completes init for the Identity extension if we can force sync and determines if we need to share state
+    /// Completes init for the Identity extension and determines if we need to share state
     /// - Parameters:
-    ///   - configSharedState: the current configuration shared state available at registration time
     ///   - event: The `Event` triggering the bootup
     /// - Returns: True if we should share state after bootup, false otherwise
-    func bootupIfReady(configSharedState: [String: Any], event: Event) -> Bool {
+    func boot(event: Event, createSharedState: ([String: Any], Event) -> Void) -> Bool {
+        if hasBooted { return true }
+
+        // load data from local storage
+        identityProperties.loadFromPersistence()
+
+        if identityProperties.ecid != nil {
+            createSharedState(identityProperties.toEventData(), event)
+            hasBooted = true
+        }
+
+        Log.debug(label: "\(LOG_TAG):\(#function)", "Identity has successfully booted up")
+
+        return hasBooted
+    }
+
+    /// Determines if there is all the required configuration and if we can force sync
+    /// - Parameters:
+    ///   - configSharedState: the current configuration shared state available at registration time
+    ///   - event: The `Event` triggering the bootup
+    /// - Returns: True if we did force synced or privacy is opted out, false otherwise
+    func forceSyncIdentifiers(configSharedState: [String: Any], event: Event) -> Bool {
         // Only bootup once we can perform the first force sync
+        if hasSynced { return true }
+
         guard readyForSyncIdentifiers(event: event, configurationSharedState: configSharedState) else {
             return false
         }
 
-        // load data from local storage
-        identityProperties.loadFromPersistence()
-
         // Load privacy status
         let privacyStr = configSharedState[IdentityConstants.Configuration.GLOBAL_CONFIG_PRIVACY] as? String ?? PrivacyStatus.unknown.rawValue
         identityProperties.privacyStatus = PrivacyStatus(rawValue: privacyStr) ?? .unknown
 
         // Update hit queue with privacy status
         hitQueue.handlePrivacyChange(status: identityProperties.privacyStatus)
 
-        hasBooted = true
-        Log.debug(label: "\(LOG_TAG):\(#function)", "Identity has successfully booted up")
         // Identity should always share its state
         // However, don't create a shared state twice, which will log an error
-        // The force sync event processed above will create a shared state if the privacy is not opt-out
-        return syncIdentifiers(event: event) != nil || identityProperties.privacyStatus == .optedOut
+        hasSynced = syncIdentifiers(event: event) != nil || identityProperties.privacyStatus == .optedOut
```

Comment:
If `hasSynced` is true, but `hasBooted` is false, then we need to create a shared state with the current IdentityProperties, event if there are none. It is important that Identity create a shared state after handling the boot event, event if that state is empty.
After creating the shared state, set `hasBooted` to true.

Reply:
Also, here `syncIdentifiers` should not create a shared state if it was already created on boot.